imhamt/src/helper.rs

```diff
@@ -1,15 +1,15 @@
 #[inline]
-pub fn clone_array_and_insert_at_pos<A: Clone>(v: &Vec<A>, a: A, pos: usize) -> Vec<A> {
+pub fn clone_array_and_insert_at_pos<A: Clone>(v: &Box<[A]>, a: A, pos: usize) -> Box<[A]> {
```

Comment:
Why does the parameter type have to be a `Box` if you clone it anyway?
I think you can just pass a slice that `Box`, `Vec`, or `Bytes` can be deref-converted to.

Reply:
ah yes, you're right. I got carried away